contracts/staking/Middleware.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/introspection/ERC165Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/extensions/AccessControlEnumerableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+import "../interfaces/staking/symbiotic/IVetoSlasher.sol";
+import "../interfaces/staking/symbiotic/IInstantSlasher.sol";
+import "../interfaces/staking/symbiotic/IVault.sol";
+import "../periphery/interfaces/IAttestationVerifier.sol";
+
+contract AttestationAutherSample is Initializable,  // initializer
+    ContextUpgradeable,  // _msgSender, _msgData
+    ERC165Upgradeable,  // supportsInterface
+    AccessControlUpgradeable,  // RBAC
+    AccessControlEnumerableUpgradeable,  // RBAC enumeration
+    UUPSUpgradeable  // public upgrade
+{
+    // in case we add more contracts in the inheritance chain
+    uint256[500] private __gap_0;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    // initializes the logic contract without any admins
+    // safeguard against takeover of the logic contract
+    constructor()
+        initializer {}
+
+    modifier onlyAdmin() {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _msgSender()), "only admin");
+        _;
+    }
+
+//-------------------------------- Overrides start --------------------------------//
+
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC165Upgradeable, AccessControlUpgradeable, AccessControlEnumerableUpgradeable) returns (bool) {
```

Comment:
I think unnecessary to inhearit `Initializable`, `ContextUpgradeable`, `ERC165Upgradeable` as `AccessControlUpgradeable` inherits them all.

Thus having only `override` should be enough?